app/src/main/java/com/willowtree/vocable/MainActivity.kt

```diff
@@ -10,129 +11,166 @@ import android.view.Surface
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Toast
-import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.children
 import androidx.core.view.isVisible
-import androidx.lifecycle.Observer
-import androidx.lifecycle.ViewModelProviders
 import com.google.ar.core.ArCoreApk
+import com.vmadalin.easypermissions.EasyPermissions
+import com.vmadalin.easypermissions.dialogs.SettingsDialog
 import com.willowtree.vocable.customviews.PointerListener
 import com.willowtree.vocable.customviews.PointerView
 import com.willowtree.vocable.databinding.ActivityMainBinding
 import com.willowtree.vocable.facetracking.FaceTrackFragment
 import com.willowtree.vocable.facetracking.FaceTrackingViewModel
-import com.willowtree.vocable.settings.SettingsViewModel
+import com.willowtree.vocable.settings.selectionmode.HeadTrackingPermissionState
+import com.willowtree.vocable.settings.selectionmode.SelectionModeViewModel
 import com.willowtree.vocable.utils.VocableSharedPreferences
 import com.willowtree.vocable.utils.VocableTextToSpeech
 import io.github.inflationx.viewpump.ViewPumpContextWrapper
 import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ViewModelOwner
+import org.koin.androidx.viewmodel.ext.android.viewModel
 import timber.log.Timber
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(),
+                     EasyPermissions.PermissionCallbacks {
 
     private val minOpenGlVersion = 3.0
     private val displayMetrics = DisplayMetrics()
     private var currentView: View? = null
-    private lateinit var viewModel: FaceTrackingViewModel
     private var paused = false
     private lateinit var binding: ActivityMainBinding
     private val sharedPrefs: VocableSharedPreferences by inject()
+    private var hasSetupAr: Boolean = false
     private val allViews = mutableListOf<View>()
-    private val settingsViewModel: SettingsViewModel by viewModels()
+
+    private val selectionModeViewModel: SelectionModeViewModel by viewModel(owner = {
+        ViewModelOwner.from(this)
+    })
+    private val faceTrackingViewModel: FaceTrackingViewModel by viewModel(owner = {
+        ViewModelOwner.from(this)
+    })
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        binding.pointerView.isVisible = false
+        setContentView(binding.root)
+
         val shouldForceDisableHeadTracking = !BuildConfig.USE_HEAD_TRACKING
         val isNotSupportedDevice = !checkIsSupportedDeviceOrFinish()
 
-        if (shouldForceDisableHeadTracking || isNotSupportedDevice) {
-            return
+        if (!shouldForceDisableHeadTracking && !isNotSupportedDevice) {
```

Comment:
It was acting as a guard previously, assuming all the AR flow requirements would be met. It was flipped because that was admittedly the quickest way around it. Will reevaluate 👍